workflow-templates/im-deploy-on-prem-website.yml

```
@@ -0,0 +1,460 @@
+      DEPLOY_BOARD_NUM: '' # TODO: Add the automated deployment board number or remove if not using an automated deployment project board.
+      DOTNET_VERSION: '' # TODO:  Set the .net core version of your project
+      PROJECT_ROOT: '' # TODO:  set this to the folder of your web site project. For most projects this will be ./src/ProjectName
+      ARTIFACT_NAME: 'Published Site' # TODO:  Set this to the name of the artifact that shows in the workflow's summary section
+      ARTIFACT_ZIP: 'artifacts.zip'
+      DEPLOY_ZIP: 'published_site.zip'
+
+    steps:
+      - name: Open a PagerDuty Maintenance Window
+        id: open-window
+        uses: im-open/open-pagerduty-maintenance-window@v1.0.0
+        with:
+          pagerduty-api-key: ${{secrets.PAGERDUTY_API_KEY}}
+          description: '${{env.PAGERDUTY_WINDOW_DESC}}'
+          minutes: ${{env.PAGERDUTY_WINDOW_IN_MIN}}
+          service-ids: ${{env.PAGERDUTY_SERVICE_IDS}}
+
+      - run: |
+          echo "The maintenance window ID is: ${{steps.open-window.outputs.maintenance-window-id}}"
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: ${{env.DOTNET_VERSION}}
+
+      # TODO:  If you need to do replacements in your appsettings.json file you can do it two ways.
+      # 1- Repeat this add-or-update-json-properties action for each environment, and add an 'if: needs.set-vars.outputs.ENVIRONMENT == ""' condition with the appropriate env to each action.
+      # 2- In the set-vars job, add one set-variable-based-on-environment action block per variable then use that variable in the single add-or-update-json-properties action below.
+      # - name: Update appsettings.json with Environment Specific Values
+      #   uses: im-open/add-or-update-json-properties@v1.0.1
+      #   with:
+      #     path-to-json-file: '${{ env.PROJECT_ROOT }}/appsettings.json'
+      #     properties-to-update-or-add: |
+      #       [
+      #         {"SecretName": "SecretValue"},
+      #         {"Nested.Secret.Name": "SecretValue"},
+      #         {"": ""}
+      #       ]
+
+      # TODO:  If you need to do any Octopus variable substitution (i.e. replacing #{OCTO_PLACEHOLDER} in files) use the following action.  Must be done once per file.
+      # - uses: im-open/octostache-action@v1.0.0
+      #   with:
+      #     variablesFile: ./substitution-variables.json                          # Files containing the variable substitutions to make, you will need to create this file
+      #     templateFile: ./src/DemoApp19/Bff/FrontEnd/scripts/build-variables.js # The file to make the substitutions in
+      #     outputFile: ./src/DemoApp19/Bff/FrontEnd/scripts/build-variables.js   # The file to save the substitutions in (generally the same as above)
```

It's like this in the other templates but we shouldn't be providing the values, we should leave it empty so it blows up if they aren't there.  I can fix it in the other places
```suggestion
      - uses: im-open/octostache-action@v1.0.0
        with:
          variablesFile: '' # TODO: Add the ./path/file containing the variable substitutions to make, you will need to create this file  See the action for more details.
          templateFile: '' # TODO:  Add the ./path/file to make the substitutions in
          outputFile: '' # TODO:  Add the ./path/file to save the substitutions in (generally the same as templateFile above)
```